all-in-one/jaeger-all-in-one-template.yml

```diff
@@ -73,6 +73,26 @@ items:
 - apiVersion: v1
   kind: Service
   metadata:
+    name: jaeger-collector
```

Comment:
Basically as you mentioned.. to make all-in-one "compatible" with prod. 

This is needed when users would like to use HTTP sender (instead of UDP), then then can point it to `http://jaeger-collector:port/api/traces`